weedcoco/repo/deposit.py

```diff
@@ -35,22 +36,26 @@ def get_all_existing_hash(repository_dir):
     }
 
 
-def setup_dataset_dir(repository_dir):
-    if not os.path.isdir(repository_dir):
-        os.mkdir(repository_dir)
-        dataset_dir = repository_dir / "dataset_1"
+def setup_dataset_dir(repository_dir, upload_id=None):
+    if upload_id is None:
+        if not os.path.isdir(repository_dir):
+            os.mkdir(repository_dir)
+            deposit_id = "dataset_1"
+        else:
+            latest_dataset_dir_index = max(
+                [
+                    int(dir.split("_")[-1])
+                    for dir in os.listdir(repository_dir)
+                    if re.fullmatch(r"^dataset_\d+$", dir)
+                ],
+                default=0,
+            )
+            deposit_id = f"dataset_{latest_dataset_dir_index + 1}"
     else:
-        latest_dataset_dir_index = max(
-            [
-                int(dir.split("_")[-1])
-                for dir in os.listdir(repository_dir)
-                if re.fullmatch(r"^dataset_\d+$", dir)
-            ],
-            default=0,
-        )
-        dataset_dir = repository_dir / f"dataset_{latest_dataset_dir_index + 1}"
+        deposit_id = upload_id
+    dataset_dir = repository_dir / deposit_id
     os.mkdir(dataset_dir)
```

Comment:
Avoid race conditions:

```suggestion
    try:
        os.mkdir(dataset_dir)
    except FileExistsError:
        if upload_id is not None:
            raise
        return setup_dataset_dir(repository_dir, upload_id=None)
```